modules/porous_flow/src/userobjects/PorousFlowCount.C

```diff
@@ -0,0 +1,108 @@
+#include "PorousFlowCount.h"
+#include "MooseMesh.h"
+
+template <>
+InputParameters
+validParams<PorousFlowCount>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+
+  MultiMooseEnum setup_options(SetupInterface::getExecuteOptions());
+  setup_options = "timestep_begin";
+  params.set<MultiMooseEnum>("execute_on") = setup_options;
+  return params;
+}
+
+PorousFlowCount::PorousFlowCount(const InputParameters & parameters)
+  : ElementUserObject(parameters), _num_upwinds(0), _num_downwinds(0), _proto_flux(0)
+{
+}
+
+PorousFlowCount::~PorousFlowCount()
+{
+  // need to free _num_upwinds, etc ?
+}
+
+void
+PorousFlowCount::initialize()
+{
+  // this gets called once at the start of each timestep,
+  // before execute()
+  _num_upwinds.resize(_mesh.nElem());
+  _num_downwinds.resize(_mesh.nElem());
+  _proto_flux.resize(_mesh.nElem());
```

Comment:
You definitely don't want to do this. First each processor only has a fraction of the total number of elements so it's already overestimating by potentially a lot. Assume that mesh IDs aren't continuous and just use a map instead.